pkg/meta/model/job_args.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"encoding/json"
+
+	"github.com/pingcap/errors"
+)
+
+// getOrDecodeArgsV2 get the argsV2 from job, if the argsV2 is nil, decode rawArgsV2
+// and fill argsV2.
+func getOrDecodeArgsV2[T JobArgs](job *Job) (T, error) {
+	if job.ArgsV2 != nil {
+		return job.ArgsV2.(T), nil
+	}
+	var v T
+	if err := json.Unmarshal(job.RawArgsV2, &v); err != nil {
+		return v, errors.Trace(err)
+	}
+	job.ArgsV2 = v
+	return v, nil
+}
+
+// JobArgs is the interface for job arguments.
+type JobArgs interface {
+	// fillJob fills the job args for submitting job. we make it private to avoid
+	// calling it directly, use Job.FillArgs to fill the job args.
+	fillJob(job *Job)
```

Comment:
Rename `fillArgs` suggested.

Reply:
from args's view, we are filling to job, see job.FillArgs